server/src/internalClusterTest/java/org/opensearch/cluster/coordination/NodeJoinLeftIT.java

```diff
@@ -0,0 +1,194 @@
+            .put(NodeConnectionsService.CLUSTER_NODE_RECONNECT_INTERVAL_SETTING.getKey(), "10s")
+            .put(FollowersChecker.FOLLOWER_CHECK_TIMEOUT_SETTING.getKey(), "200ms")
+            .put(FollowersChecker.FOLLOWER_CHECK_INTERVAL_SETTING.getKey(), "100ms")
+            .put(FollowersChecker.FOLLOWER_CHECK_RETRY_COUNT_SETTING.getKey(), 1)
+            .build();
+        // start a 3 node cluster with 1 cluster-manager
+        final String cm = internalCluster().startNode(nodeSettings);
+        internalCluster().startNode(Settings.builder().put("node.attr.color", "blue").put(nodeSettings).build());
+        final String redNodeName = internalCluster().startNode(Settings.builder().put("node.attr.color", "red").put(nodeSettings).build());
+
+        ClusterHealthResponse response = client().admin().cluster().prepareHealth().setWaitForNodes(">=3").get();
+        assertThat(response.isTimedOut(), is(false));
+
+        client().admin()
+            .indices()
+            .prepareCreate(indexName)
+            .setSettings(
+                Settings.builder()
+                    .put(IndexMetadata.INDEX_ROUTING_INCLUDE_GROUP_SETTING.getKey() + "color", "blue")
+                    .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+                    .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+            )
+            .get();
+
+        ClusterService cmClsService = internalCluster().getInstance(ClusterService.class, cm);
+        // Simulate a slow applier on the cm to delay node-left state application
+        cmClsService.addStateApplier(event -> {
+            if (event.nodesRemoved()) {
+                try {
+                    Thread.sleep(3000);
+                } catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        });
+        AtomicBoolean succeedFollowerChecker = new AtomicBoolean();
+
+        // Simulate followerchecker failure on 1 node when bb is false
+        ConnectionDelay handlingBehavior = new ConnectionDelay(() -> {
+            if (succeedFollowerChecker.get()) {
+                return;
+            }
+            try {
+                Thread.sleep(10);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+            throw new NodeHealthCheckFailureException("non writable exception");
+        });
+        MockTransportService redTransportService = (MockTransportService) internalCluster().getInstance(
+            TransportService.class,
+            redNodeName
+        );
+        redTransportService.addRequestHandlingBehavior(FOLLOWER_CHECK_ACTION_NAME, handlingBehavior);
+
+        // Loop runs 10 times to ensure race condition gets reproduced
+        for (int i = 0; i < 10; i++) {
+            // fail followerchecker by force to trigger node disconnect
+            // now followerchecker should fail and trigger node left
```

Comment:
Can this test become flaky ?  Lets also add a test to ClusterConnectionManagerTests

Reply:
Test should not become flaky because we are fixing node-join node-left logic at connection level.
And the timeout for cluster health is 60s which is more than enough time.
Just to ensure, will run this test 1000 times and confirm.

Will also add some additional coverage in ClusterConnectionManagerTests.

Reply:
ran the test 100 times, didn't see any failures. running it more times

Reply:
ran the test 500 more times, no failures